web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateSearchApiController.java

```diff
@@ -76,4 +75,207 @@
+
+            if (keywords.size() == 1) {
+                // Blur search gene
+                result.addAll(convertGene(GeneUtils.searchGene(keywords.get(0), false), keywords.get(0)));
+
+                // Blur search variant
+                result.addAll(convertVariant(AlterationUtils.lookupVariant(keywords.get(0), false, AlterationUtils.getAllAlterations()), keywords.get(0)));
+            } else if (keywords.size() == 2) {
+                // Assume one of the keyword is gene
+                Map<String, Set<Gene>> map = new HashedMap();
+                for (String keyword : keywords) {
+                    map.put(keyword, GeneUtils.searchGene(keyword, true));
+                }
+
+                //Find exact match
+                result.addAll(getMatch(map, keywords, true));
+                result.addAll(getMatch(map, keywords, false));
+
+                // If there is no match in OncoKB database, still try to annotate variant
+                if (result.size() == 0) {
+                    for (Map.Entry<String, Set<Gene>> entry : map.entrySet()) {
+                        if (entry.getValue().size() > 0) {
+                            for (Gene gene : entry.getValue()) {
+                                for (String keyword : keywords) {
+                                    if (!keyword.equals(entry.getKey())) {
+                                        Alteration alteration =
+                                            AlterationUtils.getAlteration(gene.getHugoSymbol(), keyword, null, null, null, null);
+                                        AlterationUtils.annotateAlteration(alteration, keyword);
+                                        TypeaheadSearchResp typeaheadSearchResp = newTypeaheadVariant(alteration);
+                                        typeaheadSearchResp.setVariantExist(false);
+                                        result.add(typeaheadSearchResp);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return new ResponseEntity<>(getLimit(result, limit), HttpStatus.OK);
+    }
+
+    private LinkedHashSet<TypeaheadSearchResp> getMatch(Map<String, Set<Gene>> map, List<String> keywords, Boolean exactMatch) {
+        LinkedHashSet<TypeaheadSearchResp> result = new LinkedHashSet<>();
+        if (map == null || keywords == null) {
+            return result;
+        }
+        if (exactMatch == null)
+            exactMatch = false;
+        for (Map.Entry<String, Set<Gene>> entry : map.entrySet()) {
+            if (entry.getValue().size() > 0) {
+                for (Gene gene : entry.getValue()) {
+                    Set<Alteration> alterations = AlterationUtils.getAllAlterations(gene);
+                    for (String keyword : keywords) {
+                        if (!keyword.equals(entry.getKey()))
```

Comment:
Keywords normally contain Gene + Variant(for now, we only deal with maximum two keywords). You pass in the matched genes for each keyword. When you loop through the map, you only query other keywords except the one you used to find genes.